demonstrations/tutorial_how_to_collect_mcm_stats.py

```diff
@@ -0,0 +1,218 @@
+r"""How to collect statistics of mid-circuit measurements
+=========================================================
+
+Measuring qubits in the middle of a quantum circuit execution can be useful in many ways.
+From understanding the inner workings of a circuit, hardware characterization,
+modeling and error mitigation, to error correction, algorithmic improvements and even up to full
+computations encoded as measurements in measurement-based quantum computation (MBQC).
+
+Before turning to any of these advanced topics, it is worthwhile to familiarize ourselves with
+the syntax and features around mid-circuit measurements (MCMs). In this how-to, we will focus on
+extracting statistics about measurements that are performed while a quantum circuit is up and
+running --- mid-circuit measurement statistics!
+
+.. figure:: ../_static/demonstration_assets/how_to_collect_mcm_stats/socialthumbnail_large_how_to_collect_mcm_stats.png
+    :align: center
+    :width: 50%
+
+
+"""
+
+######################################################################
+# Defining the circuit ansatz
+# ---------------------------
+#
+# We start with standard imports, setting a randomness seed, and specifying wires. As we
+# will treat the first wire different than all other wires, we define it as separate variable.
+#
+
+import pennylane as qml
+import numpy as np
+
+np.random.seed(511)
+
+first_wire = 0
+other_wires = [1, 2, 3]
+num_wires = 1 + len(other_wires)
+
+######################################################################
+# Now we create a quantum circuit ansatz that switches between a layer of simple rotation gates
+# (:class:`~.pennylane.RX`), mid-circuit measurements(:func:`~.pennylane.measure`), and a layer
+# of entangling two-qubit gates (:class:`~.pennylane.CNOT`) between the first and all other qubits.
+#
+
+
+def ansatz(x):
+    mcms = []
+
+    # Rotate all qubits
+    for w, x_ in enumerate(x):
+        qml.RX(x_, w)
+
+    # Measure first qubit
+    mcms.append(qml.measure(first_wire))
+
+    # Entangle all qubits with first qubit
+    for w in other_wires:
+        qml.CNOT([first_wire, w])
+
+    # Measure and reset all qubits but the first
+    for w in other_wires:
+        mcms.append(qml.measure(w))
+
+    return mcms
+
+
+######################################################################
+# A quantum circuit with basic MCM statistics
+# -------------------------------------------
+#
+# Before we post-process the mid-circuit measurements in this ansatz or expand the ansatz itself,
+# let's construct a simple :class:`~.pennylane.QNode` and look at the statistics of the four
+# performed MCMs; We compute the probability vector for the MCM on the first qubit and count the
+# bit strings sampled from the other three MCMs. To implement the ``QNode``, we also define a
+# shot-based qubit device.
+#
+
+dev = qml.device("default.qubit", shots=100)
+
+
+@qml.qnode(dev)
+def simple_node(x):
+    mcm1, *mcms2 = ansatz(x)
```

Comment:
```suggestion
    # apply the ansatz, and collect mid-circuit measurements.
    # here, mcm1 is the measurement of wire 0, and mcm2 is
    # a list of measurements of the remaining wires.
    mcm1, *mcms2 = ansatz(x)
```
optional suggestion, wasn't entirely clear to me

Reply:
Included locally.

Reply:
I am not used to this notation but it simplifies things a lot 👌